src/lib/validate-forms.js

```diff
@@ -0,0 +1,64 @@
+const api = require('./api')();
+const argsFormFilter = require('./args-form-filter');
+const fs = require('./sync-fs');
+const log = require('./log');
+const {
+  getFormDir,
+  getFormFilePaths,
+  formHasInstanceId
+} = require('./forms-utils');
+
+module.exports = async (projectDir, subDirectory, options={}) => {
+
+  const formsDir = getFormDir(projectDir, subDirectory);
+  if(!formsDir) {
+    log.info(`Forms dir not found: ${formsDir}`);
+    return;
+  }
+
+  let idValidationsPassed = true;
+  let validateFormsPassed = true;
+
+  const fileNames = argsFormFilter(formsDir, '.xml', options);
+  for (const fileName of fileNames) {
+    log.info(`Validating form: ${fileName}…`);
+
+    const { xformPath, filePath } = getFormFilePaths(formsDir, fileName);
+    const xml = fs.read(xformPath);
+
+    if(!formHasInstanceId(xml)) {
+      log.error(`Form at ${xformPath} appears to be missing <meta><instanceID/></meta> node. This form will not work on medic-webapp.`);
+      idValidationsPassed = false;
+      continue;
+    }
+    try {
+      await _formsValidate(xml);
+    } catch (err) {
+      log.error(`Form ${filePath} with errors, API validations response: ${err.message}`);
```

Comment:
Nit: small English grammar change, plus some additional suggestions.

```suggestion
      log.error(`Error found while validating "${filePath}". Validation response: ${err.message}`);
```